src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -225,38 +207,64 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
                 if (string.IsNullOrEmpty(container.ContainerNetwork)) // create network when Windows support it.
                 {
                     // Create local docker network for this job to avoid port conflict when multiple agents run on same machine.
-                    container.ContainerNetwork = $"vsts_network_{Guid.NewGuid().ToString("N")}";
-                    int networkExitCode = await _dockerManger.DockerNetworkCreate(executionContext, container.ContainerNetwork);
-                    if (networkExitCode != 0)
+                    // All containers within a job join the same network
+                    var existingNetwork = executionContext.Variables.Get(Constants.Variables.Agent.ContainerNetwork);
```

Comment:
This should probably be factored out to StartMultipleContainersAsync since it only needs to run once per job

Reply:
+1

Reply:
Done